udt/unlock_single_cell.rb

```diff
@@ -107,14 +107,14 @@ def hex_to_bin(s)
     CKB.load_script_hash(i, CKB::Source::INPUT, CKB::Category::LOCK) == current_lock_hash &&
       CKB.load_script_hash(i, CKB::Source::INPUT, CKB::Category::TYPE) == current_contract_hash
   end
-  if input_matches.length > 1
+  if input_matches.length != 1
```

Comment:
cause `input_index = input_matches[0]` at L120

Reply:
It's just an extra guarding so we can fail early when no matches are found.

Reply:
Yes, because I am not sure what will occur if there are no matched inputs and outputs. In the comment we have said that `The tx only has one input matching current lock hash and contract hash`, I think more explicit will be better.

Reply:
Oh yes, I misunderstood the change here 😂